# MoMA Collection — Initial Exploration

First look at the raw `Artworks.csv`: shape, dtypes, missingness, and a few sanity checks before any cleaning.

In [1]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', 40)

df = pd.read_csv('../data/raw/Artworks.csv', encoding='utf-8-sig', low_memory=False)
df.shape

(160705, 30)

In [2]:
df.head()

,Title,Artist,ConstituentID,ArtistBio,Nationality,BeginDate,EndDate,Gender,Date,Medium,Dimensions,CreditLine,AccessionNumber,Classification,Department,DateAcquired,Cataloged,ObjectID,URL,ImageURL,OnView,Circumference (cm),Depth (cm),Diameter (cm),Height (cm),Length (cm),Weight (kg),Width (cm),Seat Height (cm),Duration (sec.)
0,"Ferdinandsbrücke Project, Vienna, Austria (Ele...",Otto Wagner,6210,"(Austrian, 1841–1918)",(Austrian),(1841),(1918),(male),1896,Ink and cut-and-pasted painted pages on paper,"19 1/8 x 66 1/2"" (48.6 x 168.9 cm)",Fractional and promised gift of Jo Carole and ...,885.1996,Architecture,Architecture & Design,1996-04-09,Y,2,https://www.moma.org/collection/works/2,https://www.moma.org/media/W1siZiIsIjUyNzc3MCJ...,NaN,NaN,NaN,NaN,48.6000,NaN,NaN,168.9000,NaN,NaN
1,"City of Music, National Superior Conservatory ...",Christian de Portzamparc,7470,"(French, born 1944)",(French),(1944),(0),(male),1987,Paint and colored pencil on print,"16 x 11 3/4"" (40.6 x 29.8 cm)",Gift of the architect in honor of Lily Auchinc...,1.1995,Architecture,Architecture & Design,1995-01-17,Y,3,https://www.moma.org/collection/works/3,https://www.moma.org/media/W1siZiIsIjUyNzM3NCJ...,NaN,NaN,NaN,NaN,40.6401,NaN,NaN,29.8451,NaN,NaN
2,"Villa project, outside Vienna, Austria (Elevat...",Emil Hoppe,7605,"(Austrian, 1876–1957)",(Austrian),(1876),(1957),(male),1903,"Graphite, pen, color pencil, ink, and gouache ...","13 1/2 x 12 1/2"" (34.3 x 31.8 cm)",Gift of Jo Carole and Ronald S. Lauder,1.1997,Architecture,Architecture & Design,1997-01-15,Y,4,https://www.moma.org/collection/works/4,https://www.moma.org/media/W1siZiIsIjUyNzM3NSJ...,NaN,NaN,NaN,NaN,34.3000,NaN,NaN,31.8000,NaN,NaN
3,"The Manhattan Transcripts Project, New York, N...",Bernard Tschumi,7056,"(French and Swiss, born Switzerland 1944)",(),(1944),(0),(male),1980,Photographic reproduction with colored synthet...,"20 x 20"" (50.8 x 50.8 cm)",Purchase and partial gift of the architect in ...,2.1995,Architecture,Architecture & Design,1995-01-17,Y,5,https://www.moma.org/collection/works/5,https://www.moma.org/media/W1siZiIsIjUyNzQ3NCJ...,NaN,NaN,NaN,NaN,50.8000,NaN,NaN,50.8000,NaN,NaN
4,"Villa project, outside Vienna, Austria (Exteri...",Emil Hoppe,7605,"(Austrian, 1876–1957)",(Austrian),(1876),(1957),(male),1903,"Graphite, color pencil, ink, and gouache on tr...","15 1/8 x 7 1/2"" (38.4 x 19.1 cm)",Gift of Jo Carole and Ronald S. Lauder,2.1997,Architecture,Architecture & Design,1997-01-15,Y,6,https://www.moma.org/collection/works/6,https://www.moma.org/media/W1siZiIsIjUyNzQ3NSJ...,NaN,NaN,NaN,NaN,38.4000,NaN,NaN,19.1000,NaN,NaN


In [3]:
df.dtypes

Title                     str
Artist                    str
ConstituentID             str
ArtistBio                 str
Nationality               str
BeginDate                 str
EndDate                   str
Gender                    str
Date                      str
Medium                    str
Dimensions                str
CreditLine                str
AccessionNumber           str
Classification            str
Department                str
DateAcquired              str
Cataloged                 str
ObjectID                int64
URL                       str
ImageURL                  str
OnView                    str
Circumference (cm)    float64
Depth (cm)            float64
Diameter (cm)         float64
Height (cm)           float64
Length (cm)           float64
Weight (kg)           float64
Width (cm)            float64
Seat Height (cm)      float64
Duration (sec.)       float64
dtype: object

In [4]:
# Missingness by column
missing = df.isna().mean().sort_values(ascending=False) * 100
missing.round(1)

Seat Height (cm)      100.0
Circumference (cm)    100.0
Weight (kg)            99.8
Length (cm)            99.5
OnView                 99.2
Diameter (cm)          99.1
Duration (sec.)        98.8
Depth (cm)             88.3
ImageURL               40.7
URL                    35.3
Width (cm)             19.5
Height (cm)            19.0
Medium                  5.5
Dimensions              5.2
ArtistBio               4.2
DateAcquired            3.4
Date                    1.3
CreditLine              0.8
Artist                  0.8
Gender                  0.8
EndDate                 0.8
BeginDate               0.8
Nationality             0.8
ConstituentID           0.8
Title                   0.0
Classification          0.0
Cataloged               0.0
ObjectID                0.0
Department              0.0
AccessionNumber         0.0
dtype: float64

In [5]:
df['Classification'].value_counts().head(15)

Classification
Photograph                    35050
Print                         33373
Illustrated Book              28352
Mies van der Rohe Archive     17092
Drawing                       14369
Design                        12041
Architecture                   4568
Painting                       2453
Video                          2445
Notebook                       2003
Sculpture                      1809
Multiple                       1190
Periodical                      962
Installation                    926
Frank Lloyd Wright Archive      869
Name: count, dtype: int64

In [6]:
df['Department'].value_counts()

Department
Drawings & Prints                        82220
Architecture & Design                    34539
Photography                              33825
Painting & Sculpture                      4093
Media and Performance                     3295
Fluxus Collection                         1683
Film                                      1019
Architecture & Design - Image Archive       31
Name: count, dtype: int64

In [7]:
# DateAcquired range and parse-ability
df['DateAcquired'] = pd.to_datetime(df['DateAcquired'], errors='coerce')
df['DateAcquired'].min(), df['DateAcquired'].max(), df['DateAcquired'].isna().sum()

(Timestamp('1929-11-19 00:00:00'),
 Timestamp('2026-06-01 00:00:00'),
 np.int64(5468))

In [8]:
df['Gender'].value_counts(dropna=False).head(20)

Gender
(male)                                                  123606
(female)                                                 20247
()                                                        7363
(male) (male)                                             1755
NaN                                                       1258
(male) (male) (male)                                      1066
(male) (female)                                            963
(male) ()                                                  896
() (male)                                                  515
(female) (male)                                            425
(female) (female)                                          177
() ()                                                      161
(male) (male) (female)                                     119
(female) (male) (female) (male) (male) (male)              109
(male) (male) ()                                           100
(female) (male) (male) (male) ()                

## Notes

- `Gender`/`Nationality`/`BeginDate`/`EndDate` come wrapped in parentheses, e.g. `(male)`, `(1841)` — will need stripping.
- `Date` (creation year) is free text in places (ranges, "c. 1920", etc.) — needs a parsing pass to get a usable numeric year.
- Several artworks have multiple artists in one `Artist`/`Nationality`/`Gender` cell, separated by commas — worth checking before aggregating by artist demographics.